src/DecFP.jl

```
@@ -498,15 +498,30 @@ end # widths w
 
 Base.round(x::DecimalFloatingPoint, ::RoundingMode{:FromZero}) = signbit(x) ? floor(x) : ceil(x)
 
-Base.trunc(::Type{Integer}, x::DecimalFloatingPoint) = trunc(Int, x)
-Base.floor(::Type{Integer}, x::DecimalFloatingPoint) = floor(Int, x)
-Base.ceil(::Type{Integer}, x::DecimalFloatingPoint) = ceil(Int, x)
-Base.round(::Type{Integer}, x::DecimalFloatingPoint) = round(Int, x)
-Base.round(::Type{Integer}, x::DecimalFloatingPoint, ::RoundingMode{:NearestTiesAway}) = round(Int, x, RoundNearestTiesAway)
-Base.convert(::Type{Integer}, x::DecimalFloatingPoint) = convert(Int, x)
+for (f) in (:trunc, :floor, :ceil)
+    @eval function Base.$f(::Type{I}, x::DecimalFloatingPoint) where {I<:Integer}
+        I′ = isabstracttype(I) ? (I <: Unsigned ? UInt : Int) : I
```

On second thought, that's not right either, since Int is not necessarily a subtype of an arbitrary abstract Integer subtype.

Seems like it would be better just to use I (no I′, and let it throw a MethodError for typemin and typemax if I i s an abstract type that doesn't support those.

---

Similarly below.

---

I think the current behavior is correct.  Here are examples with `Float64`
```
julia> typeof(trunc(Integer, 7.2))
Int64

julia> typeof(trunc(Unsigned, 7.2))
UInt64

julia> typeof(trunc(Signed, 7.2))
Int64

julia> trunc(Integer, 1e30)
ERROR: InexactError: trunc(Int64, 1.0e30)
```
On a 32-bit system these are all `Int32`s.

---

If I define
```
abstract type MyInteger <: Signed; end
```
then you will still get `trunc(MyInteger, 7.2) isa Int`, which isn't correct since `Int` is not a subtype of `MyInteger`.